src/common/grpc.rs

```diff
@@ -301,6 +304,18 @@ impl GrpcServer {
         self.encode_message(resp)
     }
 
+    fn sensor_get_readings(&mut self, message: &[u8]) -> anyhow::Result<()> {
+        let req = component::sensor::v1::GetReadingsRequest::decode(message)?;
+        let sensor = match self.robot.lock().unwrap().get_sensor_by_name(req.name) {
+            Some(b) => b,
+            None => return Err(anyhow::anyhow!("resource not found")),
+        };
+
+        let readings = sensor.lock().unwrap().get_readings()?;
+        let resp = component::sensor::v1::GetReadingsResponse { readings };
```

Comment:
This works only because the mapped type in the `HashMap` that we are returning from `Sensor::get_readings` is using `prost_types::Value`. If we change it to `f64` this doesn't work, because Rust doesn't seem to automatically convert `HashMap<String, f64>` to `HashMap<String, prost_types::Value>` (not unreasonable). I'd prefer to change to that, but I was not sure what the right way to do the conversion was. I could obviously make a new `HashMap` and iterate and convert, but I'm guessing there is a better way.

Reply:
A rusty way to do it can look like this : https://gist.github.com/rust-play/a548b9e2b360a0c512d906b24a6f54af

Reply:
I'm resolving this comment as I think the `GenericReadingsResult` mechanism has addressed it.